packages/esm-patient-medications-app/src/add-drug-order/drug-order-form.component.tsx

```diff
@@ -51,10 +60,32 @@ const comboSchema = {
 };
 
 const schemaFields = {
-  freeTextDosage: z.string().refine((value) => value !== '', { message: 'Add free dosage note' }),
-  dosage: z.number({ invalid_type_error: 'A dosage is required' }),
-  unit: z.object({ ...comboSchema }, { invalid_type_error: 'Please select a unit' }),
-  route: z.object({ ...comboSchema }, { invalid_type_error: 'Please select a route' }),
+  // t( 'freeDosageErrorMessage', 'Add free dosage note')
+  freeTextDosage: z.string().refine((value) => value !== '', {
+    message: translateFrom(moduleName, 'freeDosageErrorMessage', 'Add free dosage note'),
+  }),
+
+  // t( 'dosageRequiredErrorMessage', 'A dosage is required' )
+  dosage: z.number({
+    invalid_type_error: translateFrom(moduleName, 'dosageRequiredErrorMessage', 'A dosage is required'),
+  }),
+
+  // t( 'selectUnitErrorMessage', 'Please select a unit' )
+  unit: z.object(
+    { ...comboSchema },
+    {
+      invalid_type_error: translateFrom(moduleName, 'selectUnitErrorMessage', 'Please select a unit'),
+    },
+  ),
+
+  // t( 'selectRouteErrorMessage', 'Please select a route' )
+  route: z.object(
+    { ...comboSchema },
+    {
+      invalid_type_error: translateFrom(moduleName, 'selectRouteErrorMessage', 'Please select a route'),
+    },
+  ),
```

Comment:
Not directly related to your changes, but it appears that `invalid_type_error` is getting conflated with `required_error` here.

Per the [Zod documentation](https://zod.dev/?id=strings), `invalid_type_error` is intended to customize the error message for type mismatches. This means it will trigger an error if the data types of the provided values do not align with those specified in the schema. So in this case, we'll expect to see the error if the provided content is not a string.

However, the current error messages suggest an intent to enforce the presence of these fields, which aligns more closely with the purpose of `required_error`.